File: Programas/Python/Factorial/Factorial.py
while 'true':
    archivo = open('Tarea.txt', 'a')
    resultado = 1
    i = 1
    print('----------------------------------------------------------------------------------')
    try:
        print("Bienvenidos al sistema")
        print("1. Calcular factorial")
        print("2. Salir")
        opcion=int(input("Indique que opcion desea realizar: "))

        if opcion == 1:
            print("Cual es el valor del cual desea el factorial ")
            fac=int(input())
            if fac<0:
                print("Error, no se puede calcular el factorial de un número negativo")
            else:
                if fac==0:
                    resultado=1
                    archivo.write('El factorial de '+ str(fac)+' es'+str(resultado) + '\n')
                    archivo.close()
                    print('El factorial de ', fac,' es',resultado)
                elif fac!=0:
                    for i in range(1,fac+1):
                        resultado=resultado*i
                    archivo.write('El factorial de '+ str(fac)+' es'+str(resultado)+'\n')
                    archivo.close()
                    print('El factorial de ',fac,'es',resultado)
        elif opcion ==2:
            print("Gracias por utilizar nuestro sistema")
            break
        else:
            print('No se ingreso ninguna de las opciones vuelva a intentarlo')
    except :
        print('Se ha producido un error revisar los datos ingresados')

File: Programas/Octave/Factorial/factorial.m
while(true)
    disp ('-----------------------------------------------------------------')
    disp ('Bienvenidos a nuestro sistema');
    disp ('[1]   Calcular el Factorial');
    disp ('[2]   Salir ');
    archivo=fopen('Tarea.txt','a');
    resultado=1;
    
    try
      opcion = input("Ingrese la opcion que desea realizar: ");
        switch opcion
          case {1}
            disp('Su opcion es calcular el factorial')
            fact=input('Ingrese el número que desea calcular el factorial: ');
            decimal=round(fact);
            if((fact-decimal)!=0)
              disp("Error no se puede calcular el factorial de un numero decimal");
            else              
              if fact<0
                  disp('Error, no se puede calcular el factorial de un número negativo');
              else 
                if fact==0         
                  printf ('El factorial de %i es %i \n',fact,resultado);
                  fprintf(archivo,'El factorial de %i es %i \n',fact,resultado);
                  fclose(archivo);
                elseif fact!=0
                  for n=1:fact
                    resultado=resultado*n;
                  endfor
                printf ('El factorial de %i es %i \n',fact,resultado);
                fprintf(archivo,'El factorial de %i es %i \n',fact,resultado);
                fclose(archivo);
              endif
            endif
          endif
          case {2}
            disp('Gracias por utilizar nuestro sistema');
            
            break
          otherwise
            disp('No se ha escogido una de las opciones')
        endswitch
    catch
      disp('Error los valores ingresados no son permitidos')
    end
    
endwhile
